### PR TITLE
build: Support docker to build binary package

### PR DESCRIPTION
**Pull request by rzr**
Change-Id: I79000463955d16eec8809c870f7d02c60da53432
Bug: https://github.com/mapbox/node-sqlite3/issues/418
Forwarded: https://github.com/mapbox/node-sqlite3/pull/1028
Origin: https://github.com/tizenteam/node-sqlite3
Signed-off-by: Philippe Coval <p.coval@samsung.com>

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Standard artifacts
+*.swp
+.DS_Store
+.nyc_output/
+.cache-loader/
+
+# Build artifacts
+/build
+node_modules/
+static/js/lib/stm_web.min.js
```
Comment (kewde) [inline .dockerignore]: Nit: is this required?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Run-time artifacts
+.node-persist/
+config/local.js
+static/uploads
```
Comment (kewde) [inline .dockerignore]: Same with some of these artifacts?

**Comment by kewde** [comment]
@rzr Thank you for the contributions.
Technically the PR is alright, it just seems to have some things in it that aren't necessary?
Could you trim it a bit? I look forward to adding CI support.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+RUN echo "#log: ${project}: Preparing sources" \
+  && set -x \
+  && which npm || . ~/.bashrc \
+  && npm run pack \
+  && npm pack \
+  && find build/stage/ -type f \
+  && sync
+
+WORKDIR /usr/local/${project}/${project}
+RUN echo "#log: ${project}: Installing sources" \
```
Comment (kewde) [inline Dockerfile]: I don't think this step is required?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+RUN echo "#log: ${project}: Preparing sources" \
+  && set -x \
+  && which npm || . ~/.bashrc \
+  && npm install || cat npm-debug.log \
+  && npm install \
+  && npm install --unsafe-perm --build-from-source \
+  && sync
+
+WORKDIR /usr/local/${project}/${project}
+RUN echo "#log: ${project}: Preparing sources" \
```
Comment (kewde) [inline Dockerfile]: Preparing source -> Building source?

**Comment by kewde** [comment]
Thank you @rzr

**Comment by tkurki** [comment]
Now that this is merged what's the next step? How do you use this? My ultimate goal is #418.

**Pull request merged**